### ISSUE TITLE
Duplicate glyphs in GDEF mark set

### ISSUE THREAD

**Issue by robmck-ms** [googlei18n/glyphsLib#89]
When building Selawik from Glyphs sources using fontmake, a number of entries are duplicated in the mark sets in GDEF. Not sure if this causes problems in any OS, but it doesn't conform to the OT spec, and isn't very efficient. 

Source: [selawik-test-kern.zip](https://github.com/googlei18n/glyphsLib/files/657881/selawik-test-kern.zip)

From ttx:

```
    <MarkGlyphSetsDef>
      <MarkSetTableFormat value="1"/>
      <!-- MarkSetCount=4 -->
      <Coverage index="0" Format="1">
        <Glyph value="uni0104"/>
        <Glyph value="uni0173"/>
        <Glyph value="uni0326"/>
        <Glyph value="uni0326"/>
        <Glyph value="uni0327"/>
        <Glyph value="uni0327"/>
        <Glyph value="uni0326.case"/>
        <Glyph value="uni0326.case"/>
        <Glyph value="uni0327.case"/>
        <Glyph value="uni0327.case"/>
      </Coverage>
      <Coverage index="1" Format="1">
        <Glyph value="uni0104"/>
        <Glyph value="uni0104"/>
        <Glyph value="uni0173"/>
        <Glyph value="uni0173"/>
        <Glyph value="uni0328"/>
        <Glyph value="uni0328.case"/>
      </Coverage>
      <Coverage index="2" Format="1">
        <Glyph value="uni0104"/>
        <Glyph value="uni0173"/>
        <Glyph value="uni0308"/>
        <Glyph value="uni0308"/>
        <Glyph value="uni0307"/>
        <Glyph value="uni0307"/>
        <Glyph value="uni0300"/>
        <Glyph value="uni0300"/>
        <Glyph value="uni0301"/>
        <Glyph value="uni0301"/>
        <Glyph value="uni030B"/>
        <Glyph value="uni0302"/>
        <Glyph value="uni0302"/>
        <Glyph value="uni030C"/>
        <Glyph value="uni030C"/>
        <Glyph value="uni0306"/>
        <Glyph value="uni0306"/>
        <Glyph value="uni030A"/>
        <Glyph value="uni030A"/>
        <Glyph value="uni0303"/>
        <Glyph value="uni0303"/>
        <Glyph value="uni0304"/>
        <Glyph value="uni0304"/>
        <Glyph value="uni0312"/>
        <Glyph value="uni0312"/>
        <Glyph value="uni0300.case"/>
        <Glyph value="uni0300.case"/>
        <Glyph value="uni0301.case"/>
        <Glyph value="uni0301.case"/>
        <Glyph value="uni030B.case"/>
        <Glyph value="uni0302.case"/>
        <Glyph value="uni0302.case"/>
        <Glyph value="uni030C.case"/>
        <Glyph value="uni030C.case"/>
        <Glyph value="uni0306.case"/>
        <Glyph value="uni0306.case"/>
        <Glyph value="uni030A.case"/>
        <Glyph value="uni030A.case"/>
        <Glyph value="uni0304.case"/>
        <Glyph value="uni0304.case"/>
      </Coverage>
      <Coverage index="3" Format="1">
        <Glyph value="uni0173"/>
        <Glyph value="uni030C.alt"/>
      </Coverage>
    </MarkGlyphSetsDef>

```

**Comment by anthrotype** [comment]
this looks more like a fonttools' feaLib issue than a glyphsLib one

**Comment by anthrotype** [comment]
it could also be related to ufo2ft `markFeatureWriter.py`, which is what adds the `UseMarkFilteringSet` lookup flags in features.fea, which then feaLib compiles into the `MarkGlyphSetsDef` tables:

https://github.com/googlei18n/ufo2ft/issues/52
https://github.com/googlei18n/ufo2ft/pull/57

**Comment by anthrotype** [comment]
this issue should be fixed in fonttools master (#863). We shall close it as soon as we tag a new fonttools release and bump the minimum requirements in glyphsLib.
